repose-aggregator/core/repose-core/src/main/java/org/openrepose/nodeservice/ReqResHeaderService.java

```diff
@@ -17,27 +17,21 @@
  * limitations under the License.
  * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
  */
-package org.openrepose.nodeservice.response;
+package org.openrepose.nodeservice;
 
-import org.openrepose.commons.utils.StringUtilities;
-import org.openrepose.core.services.headers.common.ViaHeaderBuilder;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.LoggerFactory;
 
 import javax.servlet.http.HttpServletRequest;
 
-public class ViaResponseHeaderBuilder extends ViaHeaderBuilder {
-
-    private final String reposeVersion;
-    private final String viaReceivedBy;
-
-    public ViaResponseHeaderBuilder(String reposeVersion, String configuredViaReceivedBy) {
-        this.reposeVersion = reposeVersion;
-        this.viaReceivedBy = StringUtilities.isBlank(configuredViaReceivedBy) ? "Repose" : configuredViaReceivedBy;
-    }
-
-    @Override
-    protected String getViaValue(HttpServletRequest request) {
-        StringBuilder builder = new StringBuilder(" ");
-
-        return builder.append(viaReceivedBy).append(" (Repose/").append(reposeVersion).append(")").toString();
+public interface ReqResHeaderService {
+    default void appendProtocolVersion(final StringBuilder builder, final HttpServletRequest request) {
```

Comment:
It may not be an issue since there really shouldn't be any shared logic between the two (i.e. request and response). Checking the HTTP protocol of the request for both cases (which is what we were doing before this story) looks out of spec to me.

https://tools.ietf.org/html/rfc7230#section-5.7.1
```
5.7.1.  Via

   The "Via" header field indicates the presence of intermediate
   protocols and recipients between the user agent and the server (on
   requests) or between the origin server and the client (on responses),
   similar to the "Received" header field in email (Section 3.6.7 of
   [RFC5322]).  Via can be used for tracking message forwards, avoiding
   request loops, and identifying the protocol capabilities of senders
   along the request/response chain.

     Via = 1#( received-protocol RWS received-by [ RWS comment ] )

     received-protocol = [ protocol-name "/" ] protocol-version
                         ; see Section 6.7
----- snip -----

   For each intermediary, the received-protocol indicates the protocol
   and protocol version used by the upstream sender of the message.
   Hence, the Via field value records the advertised protocol
   capabilities of the request/response chain such that they remain
   visible to downstream recipients; this can be useful for determining
   what backwards-incompatible features might be safe to use in
   response, or within a later request, as described in Section 2.6.
```

For the response, we should be looking at the HTTP protocol of the response we get back from the origin service.